lib/smart_proxy_openscap/arf_parser.rb

```diff
@@ -13,17 +13,13 @@ def initialize(cname, policy_id, date)
       end
 
       def as_json(arf_data)
-        begin
-          file = Tempfile.new
+        decompressed = Tempfile.create do |file|
           file.write(arf_data)
-          file.rewind
-          decompressed = `bunzip2 -dc #{file.path}`
+          file.flush
+          Proxy::OpenSCAP.execute!('bunzip2', '-dc', file.path).first
         rescue => e
           logger.error e
           raise Proxy::OpenSCAP::ReportDecompressError, "Failed to decompress received report bzip, cause: #{e.message}"
-        ensure
-          file.close
-          file.unlink
         end
         arf_file = ::OpenscapParser::TestResultFile.new(decompressed)
         rules = arf_file.benchmark.rules.reduce({}) do |memo, rule|
```

Comment:
This is more out of interest. Isn't this equivalent to:
```ruby
arf_file.benchmark.rules.to_h { |rule| [rule.id, rule] }
```

Reply:
It is, with to_h being slightly more efficient

Reply:
I was having some fun with rewriting it and I think https://github.com/theforeman/smart_proxy_openscap/pull/99 is cleaning it up a bit.